aiogram/api/client/session/aiohttp.py

```diff
@@ -1,29 +1,72 @@
 from __future__ import annotations
 
-from typing import AsyncGenerator, Callable, Optional, TypeVar, cast
+from typing import AsyncGenerator, Callable, Optional, TypeVar, Tuple, Dict, Any, Union, cast
 
-from aiohttp import ClientSession, ClientTimeout, FormData
+from aiohttp import ClientSession, ClientTimeout, FormData, BasicAuth, TCPConnector
 
 from aiogram.api.methods import Request, TelegramMethod
 
 from .base import PRODUCTION, BaseSession, TelegramAPIServer
 
 T = TypeVar("T")
+_ProxyType = Union[str, Tuple[str, BasicAuth]]
 
 
-class AiohttpSession(BaseSession):
+class AiohttpSession(BaseSession[_ProxyType]):
     def __init__(
         self,
         api: TelegramAPIServer = PRODUCTION,
         json_loads: Optional[Callable] = None,
         json_dumps: Optional[Callable] = None,
+        proxy: Optional[_ProxyType] = None,
     ):
-        super(AiohttpSession, self).__init__(api=api, json_loads=json_loads, json_dumps=json_dumps)
+        super(AiohttpSession, self).__init__(
+            api=api,
+            json_loads=json_loads,
+            json_dumps=json_dumps,
+            proxy=proxy
+        )
         self._session: Optional[ClientSession] = None
+        self.cfg.connector_type = TCPConnector
+        self.cfg.connector_init = cast(Dict[str, Any], {})
```

Comment:
Please, use explicit types instead of casting:
```suggestion
        self.cfg.connector_init: Dict[str, Any] = {}
```

I'm not sure about compatibility with simple namespaces, but it avoids type errors in the future.

Also, check the possibility to add typing for connector_type, like `: Type[BaseConnector]` or `: Type[TCPConnection]`

Reply:
1. You cannot do that.
2. It infers type from `aiohttp.TCPConnector`, I'd consider annotating it to `aiohttp.BaseConnector`

Reply:
About youtrack issue: the comment says that attribute and type declaration should be in `__init__` or `__new__` methods, so it's our case

Reply:
Casting is a problem because it can hide accidental errors, e.g.:
```
new_attr = cast(int, .1)  # no error
new_attr: int = .1  # error
```

Reply:
> About youtrack issue: the comment says that attribute and type declaration should be in `__init__` or `__new__` methods, so it's our case

This is not our case. We are not in the `types.SimpleNamespace::__init__`

Reply:
So, it's definitely better to not use `SimpleNamespace` as an attribute storage

Reply:
It can be illustrated with example:
```python
from types import SimpleNamespace
from typing import cast

class AFewMethods:
    def __init__(self):
        self.namespace = SimpleNamespace()
        self.namespace.a: int = 1
        self.namespace.b = cast(int, 1)
        reveal_type(self.namespace.a) # note: Revealed type is 'builtins.int'
        reveal_type(self.namespace.b) # note: Revealed type is 'builtins.int'

    def another_method(self):
        reveal_type(self.namespace.a) # note: Revealed type is 'Any'
        reveal_type(self.namespace.b) # note: Revealed type is 'Any'
        pass
```

Reply:
The main idea behind using `SimpleNamespace` is not messing `BaseSession`'s namespace. Treat it as `Dict[str, Any]`. I found `SimpeNamespace` "nicer", because it's a facade on top of `dict`

[documentation on topic](https://docs.python.org/3/library/types.html#types.SimpleNamespace)

Reply:
As I promised, here are a few options on how to achieve typing without private attributes for each param:
```python
from typing import (
    NamedTuple,
    Optional,
    TypedDict,
)


class ConfigTypedDict(TypedDict, total=False):
    param: int
    another: str


class ExampleTypedDict:
    def __init__(self):
        self.config: ConfigTypedDict = {}
        self.config['param'] = 1


class ConfigNamedTuple(NamedTuple):
    param: int
    another: str

class ExampleNamedTuple:
    def __init__(self):
        # immutable, so should be replaced every time when we want a change config
        self.config = ConfigNamedTuple(param=1, another='oh')


class ConfigJustClass:
    def __init__(self, param: int, another: str):
        self.param = param
        self.another = another

class ExampleJustClass:
    def __init__(self):
        self.config = ConfigJustClass(param=1, another='oh')
        self.config.param = 2
```